src/basic/slow-code.h

```diff
@@ -0,0 +1,34 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#pragma once
+
+#include "log.h"
+#include "string-util.h"
+#include "time-util.h"
+
+#include "macro.h"
+
+typedef struct slow_code_detector {
+        uint32_t timeout_sec;
+        uint32_t count;
```

Comment:
this count doesn't count anything. it's a boolean really.

and quite frnakly i'd drop that, and instead invalidate the whole structure once the for() loop iterated once, i.e. set all fields to NULL, and then exit the loop once they are (i.e. loop as long as `.file` is non_NULL, basically, exit once it is NULL)

Reply:
We can't set `.file` to NULL, because we will use `.file` in `slow_code()` later.